### ISSUE TITLE
TS warnings in 1.0.0-beta.1

### ISSUE THREAD

**Issue by ShMcK** [rematch/rematch#427]
There are some TS warnings that need to be cleaned up before the next release.

To replicate:

1. `npm run build` in the root
2. install and run `tsc` in the "examples/ts/count" folder

```bash
src/models/dolphins.ts:7:37 - error TS2345: Argument of type '{ state: number; reducers: { increment: (state: number) => number; }; effects: (dispatch: Rematch...' is not assignable to parameter of type 'Model<any, any>'.
  Property 'name' is missing in type '{ state: number; reducers: { increment: (state: number) => number; }; effects: (dispatch: Rematch...'.

  7 export const dolphins = createModel({
                                        ~
  8  state: 0,
    ~~~~~~~~~~
...
 17  }),
    ~~~~
 18 })
    ~


src/models/sharks.ts:7:35 - error TS2345: Argument of type '{ state: number; reducers: { increment: (state: number, payload: number) => number; }; effects: (...' is not assignableto parameter of type 'Model<any, any>'.
  Property 'name' is missing in type '{ state: number; reducers: { increment: (state: number, payload: number) => number; }; effects: (...'.

  7 export const sharks = createModel({
                                      ~
  8  state: 0,
    ~~~~~~~~~~
...
 24  },
    ~~~
 25 })
    ~


src/store.ts:11:27 - error TS2345: Argument of type '{ plugins: Plugin<Models, Action<any, any>>[]; models: typeof import("/Demos/rematch...' is not assignable to parameter of type 'InitConfig<typeof import("/Demos/rematch/examples/ts/count/src/models/index")>'.
  Types of property 'plugins' are incompatible.
    Type 'import("Demos/rematch/plugins/select/node_modules/@rematch/core/src/typings").Plugi...' is not assignable to type 'rematch.Plugin<rematch.Models, rematch.Action<any, any>>[]'.
      Type 'import("Demos/rematch/plugins/select/node_modules/@rematch/core/src/typings").Plugi...' is not assignable to type 'rematch.Plugin<rematch.Models, rematch.Action<any, any>>'.
        Types of property 'config' are incompatible.
          Type 'import("Demos/rematch/plugins/select/node_modules/@rematch/core/src/typings").InitC...' is not assignable to type 'rematch.InitConfig<rematch.Models>'.
            Types of property 'models' are incompatible.
              Type 'import("Demos/rematch/plugins/select/node_modules/@rematch/core/src/typings").Models' is not assignable to type 'rematch.Models'.
                Index signatures are incompatible.
                  Type 'import("Demos/rematch/plugins/select/node_modules/@rematch/core/src/typings").Model...' is not assignable to type 'rematch.Model<any, any>'.
                    Property 'name' is optional in type 'Model<any, any>' but required in type 'Model<any, any>'.

11 export const store = init({
                             ~
12  plugins: [selectPlugin()],
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~
13  models,
   ~~~~~~~~
14 })
```

**Comment by ShMcK** [comment]
@brunolemos any ideas what might be causing these warnings?

**Comment by brunolemos** [comment]
@ShMcK so this happened when we split Model and ModelConfig. Fix: #430
